Mathlib/Analysis/ShiftedLegendrePoly.lean

```diff
@@ -0,0 +1,93 @@
+/-
+Copyright (c) 2025 Junqi Liu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Junqi Liu
+-/
+import Mathlib.Algebra.Polynomial.AlgebraMap
+import Mathlib.Algebra.Polynomial.Derivative
+import Mathlib.Analysis.RCLike.Basic
+
+/-!
+# shifted Legendre Polynomials
+
+In this file, we define the shifted Legendre polynomials `shiftedLegendre n` for `n : ℕ` as a
+polynomial in `ℤ[X]`. We prove some basic properties of the Legendre polynomials.
+-/
+
+open scoped Nat
+open BigOperators Finset
+
+namespace Polynomial
+
+/-- `shiftedLegendre n` is the polynomial defined as the sum of integer monomials. -/
+noncomputable def shiftedLegendre (n : ℕ) : ℤ[X] :=
+  ∑ k ∈ Finset.range (n + 1), C ((-1) ^ k * (Nat.choose n k) * (Nat.choose (n + k) n) : ℤ) * X ^ k
+
+/-- The expand of `shiftedLegendre n`. -/
+theorem mul_shiftedLegendre_n_dervi (n : ℕ) : (n ! : ℤ[X]) * (shiftedLegendre n) =
```

Comment:
We don't use `dervi` in mathlib for `derivative`. I think `factorial_mul_shiftedLegendre_eq` is a better name.

Reply:
Also the docstring "The expand of `shiftedLegendre n`" is not correct. Please modify.